src/MiniValidation/TypeDetailsCache.cs

```diff
@@ -183,16 +183,18 @@ private static (ValidationAttribute[]?, DisplayAttribute?, SkipRecursionAttribut
             }
         }
 
+        var customAttributes = new List<Attribute>();
+
         var propertyAttributes = property.GetCustomAttributes();
-        var customAttributes = paramAttributes is not null
-            ? paramAttributes.Concat(propertyAttributes)
-            : propertyAttributes;
+        customAttributes.AddRange(propertyAttributes);
+        if (paramAttributes is not null)
+        {
+            customAttributes.AddRange(paramAttributes);
+        }
 
-        if (TryGetAttributesViaTypeDescriptor(property, out var typeDescriptorAttributes))
+        if (customAttributes.Count == 0 && TryGetAttributesViaTypeDescriptor(property, out var typeDescriptorAttributes))
```

Comment:
I noticed that we were getting duplicate validation messages because the attributes were being pulled off the models in multiple ways. I am not sure what the scenarios are where getting attributes from typedescriptors would come into play, but it was ugly having multiple of the same validation message. Thoughts?